agent/consul/gateways/bind.go

```diff
@@ -71,3 +84,52 @@ func getReferences(route structs.BoundRoute) (referenceSet, gatewayRefs) {
 
 	return parentRefs, gatewayRefs
 }
+
+func requestToResourceRef(req controller.Request) structs.ResourceReference {
+	ref := structs.ResourceReference{
+		Kind: req.Kind,
+		Name: req.Name,
+	}
+	if req.Meta != nil {
+		ref.EnterpriseMeta = *req.Meta
+	}
+	return ref
+}
+
+// RemoveGateway sets the route status for the given gateway to be unbound if it should be bound
+func RemoveGateway(gateway structs.ResourceReference, entries ...structs.BoundRoute) []structs.ControlledConfigEntry {
+	now := time.Now()
```

Comment:
Do we want this to be `time.Now().UTC()`? `Now()` will return local time. So long as all servers are in the same region that would be fine. I just wonder if there is a situation where a comparison is done between servers in different regions and something doesn't work.